### PR TITLE
FMWK-425 Document FilterOperations

### PR DESCRIPTION
**Pull request by agrgr**
- Document FilterOperations
- Update logging

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java [hunk 1]
```diff
@@ -454,6 +507,10 @@ public Filter sIndexFilter(Map<QualifierKey, Object> qualifierMap) {
             return null; // not supported
         }
     },
+    /**
+     * For use in queries "find records where Map key <...> has value equal to a given object". The Map can be a whole
+     * bin or part of it (nested inside one or more Lists or other Maps). Maps in the database represent also POJOs.
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:512]: > Maps in the database represent also POJOs.

I think its not clear enough, I would change that to something like "Maps in Aerospike database can represent a Map or an entire POJO".

The same phrase is used many times in this file.

File: src/main/java/org/springframework/data/aerospike/query/StatementBuilder.java [hunk 2]
```diff
@@ -140,7 +140,7 @@ private boolean isIndexedBin(Statement stmt, Qualifier qualifier) {
         }
 
         if (log.isDebugEnabled() && hasField) {
-            log.debug("Query #{}, bin {}.{}.{} has secondary index: {}",
+            log.debug("Qualifier #{}, bin {}.{}.{} has secondary index: {}",
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/StatementBuilder.java:143]: Does printing Qualifier Hashcode has any meaning from the user's perspective? does it help us in debugging?

Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/StatementBuilder.java:143]: Hashcode serves as a unique identifier to find other log messages related to the qualifier.

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java (hunk 1 above)
Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:512]: Rephrased a bit differently

**Review by reugn — APPROVED** [review]

**Pull request merged**